operator/pkg/controller/utils/reconcile_test.go

```diff
@@ -48,6 +48,64 @@ var _ = Describe("SubReconciler tests", func() {
 			_, err := r.GetResult()
 			Expect(err).To(HaveOccurred())
 		})
+
+		It("is not superior to NothingToDo", func() {
+			r := utils.NothingToDo()
+
+			Expect(r.IsSuperior(utils.NothingToDo())).To(BeFalse())
+		})
+
+		It("is not superior to NeedDelayedRequeue", func() {
+			r := utils.NothingToDo()
+
+			Expect(r.IsSuperior(utils.NeedDelayedRequeue())).To(BeFalse())
+		})
+
+		It("is not superior to NeedRequeue", func() {
+			r := utils.NothingToDo()
+
+			Expect(r.IsSuperior(utils.NeedRequeue(nil))).To(BeFalse())
+		})
+
+		It("is not superior to EndReconciliation", func() {
+			r := utils.NothingToDo()
+
+			Expect(r.IsSuperior(utils.NeedEndReconciliation())).To(BeFalse())
+		})
```

Comment:
Yes likely (not sure it will be easier to read but at least less code duplication)

Reply:
I did not use DescribeTable since I did not find any way to have it clearer than now, but I moved to a single `It` per SubReconcilerResult type for all "superior"

See https://github.com/scality/metalk8s/pull/4000/commits/2fbab1f825034ac1e3c199008e010affe5468b0a